src/amo/components/RatingsByStar/styles.scss

```diff
@@ -32,49 +32,24 @@
 .RatingsByStar-barValue {
   background-color: rgba($yellow-50, 0.75);
 
-  // Pre-defined widths based on % of stars.
-  &.RatingsByStar-barValue--0pct {
-    width: 0%;
-  }
-
-  &.RatingsByStar-barValue--10pct {
-    width: 10%;
-  }
-
-  &.RatingsByStar-barValue--20pct {
-    width: 20%;
-  }
-
-  &.RatingsByStar-barValue--30pct {
-    width: 30%;
-  }
-
-  &.RatingsByStar-barValue--40pct {
-    width: 40%;
-  }
-
-  &.RatingsByStar-barValue--50pct {
-    width: 50%;
-  }
-
-  &.RatingsByStar-barValue--60pct {
-    width: 60%;
-  }
-
-  &.RatingsByStar-barValue--70pct {
-    width: 70%;
-  }
-
-  &.RatingsByStar-barValue--80pct {
-    width: 80%;
-  }
-
-  &.RatingsByStar-barValue--90pct {
-    width: 90%;
-  }
-
-  &.RatingsByStar-barValue--100pct {
-    width: 100%;
+  $widths: (
+    0: 0%,
+    10: 10%,
+    20: 20%,
+    30: 30%,
+    40: 40%,
+    50: 50%,
+    60: 60%,
+    70: 70%,
+    80: 80%,
+    90: 90%,
+    100: 100%,
```

Comment:
Could you use a loop with a step of 10 and reuse `width` or `i` in both places (class name and width)?